### PR TITLE
test: Add missing unit test for covering azure.ParseResourceID()

### LINKED ISSUE #3622: Add test coverage for azure.ParseResourceID()
A [codecov comment](https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3616#issuecomment-1583014546) pointed out that some large controller functions are not currently unit tested, specifically those that are calling `azure.ParseResourceID()`.

There would be some mocking and test setup involved, but it should be possible to add unit tests covering these areas.

### PR DESCRIPTION
**Pull request by chandankumar4**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Add the missing unit test case for covering azure.ParseResourceID()

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3622 

**Special notes for your reviewer**:

- [ ] cherry-pick candidate

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: azure/services/virtualmachines/client.go [hunk 1]
```diff
@@ -74,29 +71,6 @@ func (ac *AzureClient) Get(ctx context.Context, spec azure.ResourceSpecGetter) (
 	return resp.VirtualMachine, nil
 }
 
-// GetByID retrieves information about the model or instance view of a virtual machine.
-func (ac *AzureClient) GetByID(ctx context.Context, resourceID string) (armcompute.VirtualMachine, error) {
```
Comment (chandankumar4) [inline azure/services/virtualmachines/client.go:78]: This method is not used anywhere

**Review by mboersma — COMMENTED** [review]
/lgtm

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**